### DISCUSSION TITLE
issue: Checkbox double click: causes a form validation (required) error

### DISCUSSION THREAD

**Discussion by Asher978** [react-hook-form/react-hook-form#7635] [Q&A]
### Version Number

7.24.0

### Codesandbox/Expo snack

https://codesandbox.io/s/rough-sound-x5gle?file=/src/App.js

### Steps to reproduce

1. Go to 'https://codesandbox.io/s/rough-sound-x5gle?file=/src/App.js'
2. Expected behavior:
- click on `submit` without selecting checkbox > error is rendered
- any of the checkbox is selected by clicking once > no error
- same checkbox is clicked again > input loses its value > new form error

### Expected behaviour

User submits the form without selecting any of the checkboxes
Error is presented
User clicks on the checkbox
User clicks again on the same checkbox un-intentionally > error is rendered

Should be no error as the checkbox is selected, yet the error stays.

Any help will be greatly appreciated

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
hey @Asher978 this type of usage could be troublesome, which mixing with uncontrolled with controlled. Perhaps you want to pick one? I have a checkboxes example for `useController`: https://react-hook-form.com/api/usecontroller
![Screen Shot 2022-01-24 at 5 32 48 pm](https://user-images.githubusercontent.com/10513364/150733390-12400689-c6e0-4c06-a169-5642761ad6de.png)

for uncontrolled, all you need just `register` API, you shouldn't involved with `setState `

**Comment by Asher978** [comment]
What if you want one of the checkboxes to be checked out of the group using controlled?

**Reply by Moshyfawn** [reply]
Can you elaborate on what do you mean by "checked out of the group"?

**Comment by Asher978** [comment]
@Moshyfawn 

I have a group of checkboxes on a form.

Validation needs to check the following:
At minimum, one of the checkboxes must be checked, otherwise return an error

```js 
<input type='checkbox' name='roles' value="admin" />
<input type='checkbox' name='roles' value="manager" />
<input type='checkbox' name='roles' value="billing" />
```

**Reply by bluebill1049** [reply]
what about something like this: https://codesandbox.io/s/eager-danny-1pqt9?file=/src/App.js if you go with uncontrolled

```tsx
import * as React from "react";
import { useForm } from "react-hook-form";
import Headers from "./Header";
import "./styles.css";

let renderCount = 0;

export default function App() {
  const { register, handleSubmit, getValues } = useForm();
  const onSubmit = (data) => console.log(data);
  renderCount++;

  const validate = () => {
    console.log(getValues()) // doing your validation here
  }

  return (
    <div>
      <Headers
        renderCount={renderCount}
        description="Performant, flexible and extensible forms with easy-to-use validation."
      />

      <form onSubmit={handleSubmit(onSubmit)}>
        <input type="checkbox" {...register("firstName", { validate })} placeholder="First Name" />
        <input type="checkbox" {...register("firstName", { validate })} placeholder="First Name" />
        <input type="checkbox" {...register("firstName", { validate })} placeholder="First Name" />
        <input type="submit" />
      </form>
    </div>
  );
}

```

**Reply by Moshyfawn** [reply]
@bluebill1049 You CSB is empty
<img width="262" alt="image" src="https://user-images.githubusercontent.com/16290753/150976275-323badc1-0502-4e24-ad64-7593c9bae1fa.png">

**Reply by bluebill1049** [reply]
updated :) forgot to save

**Comment by Asher978** [comment]
Hi @bluebill1049 

I really appreciate your effort and willingness to help.
However, using uncontrolled still leaves me with not having a control on checkbox selected for only one option.

Ultimately, if I have a group of checkboxes,

```js

<input name="checkbox" value="admin" />
<input name="checkbox" value="manager" />
<input name="checkbox" value="billing" />
 
```

User should be ONLY able to click on of the checkboxes and not have the option to select any other if one o them is checked. In addition, validation should render an error if none of the checkboxes are selected.

**Reply by Moshyfawn** [reply]
It sounds like the [radio input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio) is what you're after, tbh

**Comment by Asher978** [comment]
My gosh ... 

it was that simple .... !!!!

Sorry for the troubles.

**Reply by Moshyfawn** [reply]
No worries! I'm glad I could help ;)
